src/ORM/ArrayLib.php

```diff
@@ -274,4 +274,22 @@ public static function iterateVolatile(array &$list)
             }
         }
     }
+
+    /**
+     * Similar to shuffle, but retains the existing keys.
+     * Shuffles the array in place.
+     */
+    public static function shuffleAssociative(array &$array): void
```

Comment:
This is needed because the keys of the new list's `$rows` array is associative, with IDs as keys.